shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/main/java/org/apache/shardingsphere/proxy/frontend/postgresql/auth/PostgreSQLMd5Digest.java

```diff
@@ -0,0 +1,94 @@
+/*
```

Comment:
We can't replace BSD-2-Clause License header with Apache 2.0 License. 
BSD-2-Clause License requests: **Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.**

The actual PasswordMessage can be computed in SQL as concat('md5', md5(concat(md5(concat(password, username)), random-salt))). (Keep in mind the md5() function returns its result as a hex string.)